koswat/cost_report/summary/koswat_summary_builder.py

```diff
@@ -94,13 +95,22 @@ def _get_final_reinforcement_per_location(
         self,
         locations_profile_report_list: list[MultiLocationProfileCostReport],
         available_locations: list[PointSurroundings],
-    ) -> dict[PointSurroundings, ReinforcementProfile]:
+    ) -> list[StrategyLocationReinforcement]:
         _matrix = KoswatSummaryLocationMatrixBuilder(
             available_locations=available_locations,
             locations_profile_report_list=locations_profile_report_list,
         ).build()
+        _reinforcements = [
+            StrategyReinforcementType(
+                reinforcement_type=type(x.profile_cost_report.reinforced_profile),
+                base_costs=x.profile_cost_report.total_cost,
+                ground_level_surface=x.profile_cost_report.reinforced_profile.new_ground_level_surface,
```

Comment:
Open question, why would we do this? We are better just setting the `reinforcement` property, then you can via a property `reinforcement_type` return directly the type. This way we keep all available data available (at the expense of memory though).

Once I finish the review I might change my opinion though.